src/helpers/buffers/fsv.rs

```diff
@@ -73,15 +74,16 @@ impl<const N: usize> FixedSizeByteVec<N> {
     /// ## Panic
     /// Panics if `ready()` is false before calling `drain()`.
     pub fn take(&mut self) -> Option<Vec<u8>> {
-        if self.added[..self.region_size].all() {
-            self.added.drain(..self.region_size).for_each(drop);
-            let r = self.data.drain(..self.region_size * N).collect();
-            self.elements_drained += self.region_size;
+        let last_inserted = self.added.leading_ones();
```

Comment:
so the main difference for this PR is that the returned `Vec<u8>` previously would always return with length `self.region_size`, whereas now, it returns length of whatever `self.added.leading_ones()` is?

Reply:
that is correct. Previously it always returned chunks of the same size, even if next chunk was ready, `fsv` required another call to `take` to take it out. Now it tries to push everything it can out which is a contiguous set of elements in the front of the queue.